### PR TITLE
MINOR: add window verification to sliding-window co-group test

### PR DESCRIPTION
**Pull request by mjsax**
Follow up to #10703 \cc @ijuma

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/SlidingWindowedCogroupedKStreamImplTest.java [hunk 1]
```diff
@@ -169,34 +167,37 @@ public void slidingWindowAggregateStreamsTest() {
...
-            expected.add(new TestRecord<>("k2", "0+A+A+B+B", null, 504L));
-            expected.add(new TestRecord<>("k1", "0+A+B+B", null, 504L));
-            expected.add(new TestRecord<>("k1", "0+B+B", null, 504L));
-            expected.add(new TestRecord<>("k1", "0+B", null, 504L));
-            expected.add(new TestRecord<>("k1", "0+A+A+B+B", null, 504L));
+            final List<TestRecord<Windowed<String>, String>> results = testOutputTopic.readRecordsToList();
+
+            final List<TestRecord<Windowed<String>, String>> expected = new LinkedList<>();
+            // k1-A-500
+            expected.add(new TestRecord<>(new Windowed<>("k1", new TimeWindow(0L, 500L)), "0+A", null, 500L));
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/SlidingWindowedCogroupedKStreamImplTest.java:174]: Just realized that using `TimeWindows` is actually incorrect... (filed https://issues.apache.org/jira/browse/KAFKA-12839)

\cc @lct45 @ableegoldman

**Review by showuon — APPROVED** [review]
@mjsax , thanks for the PR. LGTM! 

All failed tests are flaky `RaftClusterTest`.
```
    Build / JDK 15 and Scala 2.13 / kafka.server.RaftClusterTest.testCreateClusterAndCreateListDeleteTopic()
    Build / JDK 8 and Scala 2.12 / kafka.server.RaftClusterTest.testCreateClusterAndCreateAndManyTopics()
    Build / JDK 8 and Scala 2.12 / kafka.server.RaftClusterTest.testCreateClusterAndCreateAndManyTopics()
    Build / JDK 11 and Scala 2.13 / kafka.server.RaftClusterTest.testCreateClusterAndCreateListDeleteTopic()
```

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/SlidingWindowedCogroupedKStreamImplTest.java (hunk 1 above)
Comment (ableegoldman) [reply streams/src/test/java/org/apache/kafka/streams/kstream/internals/SlidingWindowedCogroupedKStreamImplTest.java:174]: >A {@link TimeWindow} covers a half-open time interval

I was about to say we should just make `TimeWindow` un-opinionated, but this is literally the first thing in the javadocs for the class. So I'd say it's pretty clear about what it's representing -- totally missed this before, I thought it was just a basic container class

**Review by ableegoldman — APPROVED** [review]
Ok with me

**Pull request merged**